### PR TITLE
Add envvars for various command-line switches in the pipeline

### PR DESCRIPTION
**Pull request by samshah**

### CODE REVIEW

File: R/scripts/filter_MC.R [hunk 1]
```diff
@@ -9,20 +9,21 @@ suppressMessages(library(magrittr))
...
+  optparse::make_option(c("-c", "--config"), action="store", default=Sys.getenv("COVID_CONFIG_PATH", Sys.getenv("CONFIG_PATH")), type='character', help="path to the config file"),
+  optparse::make_option(c("-s", "--scenarios"), action="store", default=Sys.getenv("COVID_SCENARIOS", 'all'), type='character', help="name of the intervention to run, or 'all' to run all of them"),
+  optparse::make_option(c("-d", "--deathrates"), action="store", default=Sys.getenv("COVID_DEATHRATES", 'all'), type='character', help="name of the death scenarios to run, or 'all' to run all of them"),
+  optparse::make_option(c("-j", "--jobs"), action="store", default=Sys.getenv("COVID_NJOBS", parallel::detectCores()), type='integer', help="Number of jobs to run in parallel"),
+  optparse::make_option(c("-k", "--simulations_per_slot"), action="store", default=Sys.getenv("COVID_SIMULATIONS_PER_SLOT"), type='integer', help = "number of simulations to run for this slot"),
+  optparse::make_option(c("-n", "--number_of_simulations"), action="store", default=Sys.getenv("COVID_NSIMULATIONS", 1), type='integer', help = "number of slots to run"),
```
Comment (samshah) [inline R/scripts/filter_MC.R:22]: suggestions for dealing with the confusing simulations/slot nomenclature? maybe i could tackle it here

Comment (jkamins7) [reply R/scripts/filter_MC.R:22]: i'm planning on tackling this as part of filename fix, but aprt of that can be done immediately:
 - slots are normally called chains (n->N,i->n)
 - simulations per slot is normally called iterations (k->K,new k)
 - maybe go ahead and add an unused block (new B,new b)
 - I think using different letters for # of x and index inside x is weird, and we should just do lowercase and capital
open to other suggestions

Comment (samshah) [reply R/scripts/filter_MC.R:22]: so:
- `COVID_NSIMULATIONS` -> `COVID_NCHAINS`
- `COVID_SIMULATIONS_PER_SLOT` -> `COVID_ITERATIONS`

this should apply only to the inference jobs? what about the existing seir/hosp code and the params they take?

Comment (jkamins7) [reply R/scripts/filter_MC.R:22]: that's a good point.  i think we should continue calling them simulations. So maybe COVID_NSIMULATIONS and COVID_FIRST_SIMULATION.

**Comment by samshah** [comment]
are we good to move this PR to review (cc @jwills)?

**Comment by jkamins7** [comment]
This PR depends on the filename changes.  I have another branch where those changes are happening.  Thoughts on merging this into that branch instead?

**Comment by samshah** [comment]
sounds good to me

File: R/scripts/full_filter.R [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
-  optparse::make_option(c("-s", "--scenarios"), action="store", default='all', type='character', help="name of the intervention to run, or 'all' to run all of them"),
-  optparse::make_option(c("-d", "--deathrate"), action="store", default='all', type='character', help="name of the death scenarios to run, or 'all' to run all of them"),
-  optparse::make_option(c("-j", "--jobs"), action="store", default="72", type='integer', help="Number of jobs to run in parallel"),
-  optparse::make_option(c("-k", "--sims_per_slot"), action="store", default=NA, type='integer', help = "Number of simulations to run per slot"),
-  optparse::make_option(c("-n", "--slots"), action="store", default=NA, type='integer', help = "Number of slots to run."),
+  optparse::make_option(c("-c", "--config"), action="store", default=Sys.getenv("COVID_CONFIG_PATH", Sys.getenv("CONFIG_PATH")), type='character', help="path to the config file"),
+  optparse::make_option(c("-p", "--pipepath"), action="store", type='character', help="path to the COVIDScenarioPipeline directory", default = Sys.getenv("COVID_PATH", "COVIDScenarioPipeline/")),
```
Comment (kkintaro) [inline R/scripts/full_filter.R:5]: Should this be COVID_PATH?

File: R/scripts/test_MC.R [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
-  optparse::make_option(c("-s", "--scenarios"), action="store", default='all', type='character', help="name of the intervention to run, or 'all' to run all of them"),
-  optparse::make_option(c("-d", "--deathrate"), action="store", default='all', type='character', help="name of the death scenarios to run, or 'all' to run all of them"),
-  optparse::make_option(c("-j", "--jobs"), action="store", default="72", type='integer', help="Number of jobs to run in parallel"),
-  optparse::make_option(c("-k", "--sims_per_slot"), action="store", default=NA, type='integer', help = "Number of simulations to run per slot"),
-  optparse::make_option(c("-n", "--slots"), action="store", default=NA, type='integer', help = "Number of slots to run."),
+  optparse::make_option(c("-c", "--config"), action="store", default=Sys.getenv("COVID_CONFIG_PATH", Sys.getenv("CONFIG_PATH")), type='character', help="path to the config file"),
+  optparse::make_option(c("-p", "--pipepath"), action="store", type='character', help="path to the COVIDScenarioPipeline directory", default = Sys.getenv("COVID_PATH", "COVIDScenarioPipeline/")),
```
Comment (kkintaro) [inline R/scripts/test_MC.R:5]: COVID_PATH here as well?

File: R/scripts/full_filter.R (hunk 2 above)
Comment (samshah) [reply R/scripts/full_filter.R:5]: not sure I understand, are you suggesting an alternative name vs COVID_PATH?

**Comment by jkamins7** [comment]
I've rebased this, and (i hope) resolved the conflicts without clobbering any of the environment variables.  Unfortunately, I've also added more command line arguments, some of which (at least run_id, and block_id) need to be passed to the script via environment variable.  Still in the process of testing, but I think it's at a point where we're safe to merge this in (it works in the same circumstances that it worked before one scenario/deathrate configs)

**Pull request closed without merging**